### PR TITLE
Makes the JDBC server to work

### PR DESCRIPTION
**Pull request by liancheng**
Made some minor changes to make JDBC server to work:
1. `SharkServer2` should be started with the `run` script to leverage the computed `CLASSPATH` and detect user Hive and Hadoop configurations.
2. Hive native commands in Spark SQL now have output too, so always `setHasResultSet(true)`.
3. Updated dependencies: `hive-beeline` is added as a runtime dependency required by `bin/beeline`.

A sample session below. Starting `SharkServer2`:

```
$ ./bin/shark --service sharkserver2
...
14/06/18 19:41:29 INFO shark.SharkServer2$: SharkServer2 started
14/06/18 19:41:29 INFO thrift.ThriftCLIService: ThriftBinaryCLIService listening on localhost/127.0.0.1:10000
```

Connect to the JDBC server with BeeLine:

```
$ ./bin/beeline

Beeline version 0.12.0 by Apache Hive
beeline> !connect jdbc:hive2://localhost:10000
scan complete in 3ms
Connecting to jdbc:hive2://localhost:10000
Enter username for jdbc:hive2://localhost:10000: lian
Enter password for jdbc:hive2://localhost:10000: 
Connected to: Hive (version 0.12.0)
Driver: Hive (version 0.12.0)
Transaction isolation: TRANSACTION_REPEATABLE_READ
0: jdbc:hive2://localhost:10000> show tables;
+--------------+
|    result    |
+--------------+
| hive_table   |
| shark_table  |
+--------------+
2 rows selected (0.952 seconds)
0: jdbc:hive2://localhost:10000> explain select key, count(*) from shark_table group by key limit 10;
+-------------------------------------------------------------------------------------+
|                                        plan                                         |
+-------------------------------------------------------------------------------------+
| ExplainCommand [plan#44:0]                                                          |
|  Limit 10                                                                           |
|   Aggregate false, [key#37], [key#37,SUM(PartialCount#39L) AS c_1#35L]              |
|    Exchange (HashPartitioning [key#37:0], 200)                                      |
|     Exchange (HashPartitioning [key#37:0], 200)                                     |
|      Aggregate true, [key#37], [key#37,COUNT(1) AS PartialCount#39L]                |
|       HiveTableScan [key#37], (MetastoreRelation default, shark_table, None), None  |
+-------------------------------------------------------------------------------------+
7 rows selected (0.109 seconds)
0: jdbc:hive2://localhost:10000> select key, count(*) from shark_table group by key limit 10;        
+------+------+
| key  | c_1  |
+------+------+
| 152  | 2    |
| 76   | 2    |
| 128  | 3    |
| 311  | 3    |
| 316  | 3    |
| 435  | 1    |
| 323  | 1    |
| 27   | 1    |
| 190  | 1    |
| 406  | 4    |
+------+------+
10 rows selected (1.91 seconds)
0: jdbc:hive2://localhost:10000>
```

### CODE REVIEW

File: project/SharkBuild.scala [hunk 1]
```diff
@@ -32,7 +32,7 @@ object SharkBuild extends Build {
 
   val SHARK_ORGANIZATION = "edu.berkeley.cs.shark"
 
-  val SPARK_VERSION = "1.0.0-SNAPSHOT"
+  val SPARK_VERSION = "1.1.0-SNAPSHOT"
```
Comment (pwendell) [inline project/SharkBuild.scala]: Could this actually just link against spark 1.0.0?

**Comment by pwendell** [comment]
Hey @liancheng I tried to run this from a totally fresh clone of your repo and I got a dependency issue (although - I did build against spark 1.0.0 instead of a snapshot, maybe that is the problem?).

```
$ sbt/sbt package
./bin/shark --service sharkserver2
14/06/18 20:35:30 ERROR actor.ActorSystemImpl: Uncaught fatal error from thread [spark-akka.actor.default-dispatcher-4] shutting down ActorSystem [spark]
java.lang.VerifyError: (class: org/jboss/netty/channel/socket/nio/NioWorkerPool, method: createWorker signature: (Ljava/util/concurrent/Executor;)Lorg/jboss/netty/channel/socket/nio/AbstractNioWorker;) Wrong return type in function
    at akka.remote.transport.netty.NettyTransport.<init>(NettyTransport.scala:282)
    at akka.remote.transport.netty.NettyTransport.<init>(NettyTransport.scala:239)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
```

**Comment by liancheng** [comment]
Yea, we need 1.1.0-SNAPSHOT to run the JDBC server.

**Comment by elyast** [comment]
To be honest I was able to run shark server against spark 1.0.0 (couple of days ago) without using sbt however there was a need to tweak dependencies, so I think at least it should be possible.

I would hate to wait for spark 1.1 in order to run shark

File: project/SharkBuild.scala (hunk 1 above)
Comment (liancheng) [inline project/SharkBuild.scala]: No, Hive native commands like `SHOW TABLES` and `EXPLAIN` doesn't work properly. But confirmed that 1.0.1-SNAPSHOT is good enough. Will update this soon.

**Comment by liancheng** [comment]
Hi @elyast Yea, I understand :-) With 1.0.0, Hive native commands like `SHOW TABLES` and `EXPLAIN` cannot work properly, but 1.0.1-SNAPSHOT is good enough. Hopefully 1.0.1 will be released much sooner than 1.1.0.

**Comment by liancheng** [comment]
@pwendell Jenkins fails because org.apache.spark#spark-hive_2.10;1.0.1-SNAPSHOT not found. Need to do a publish-local on Jenkins node?

**Comment by elyast** [comment]
@liancheng indeed SHOW TABLES doesn't work correctly :), for me it was minor problem, however 1.0.1 does sound better, thanks

**Pull request merged**